contracts/pools/weighted/WeightedPool2Tokens.sol

```diff
@@ -185,20 +188,29 @@ contract WeightedPool2Tokens is
         emit SwapFeePercentageChanged(swapFeePercentage);
     }
 
-    function _isOracleEnabled() private view returns (bool) {
-        return _miscData.decodeBool(_MISC_ORACLE_ENABLED_OFFSET);
+    function enableOracle() external authenticate {
+        _setOracleEnabled(true);
     }
 
     function _setOracleEnabled(bool enabled) private {
         _miscData = _miscData.storeBoolean(true, _MISC_ORACLE_ENABLED_OFFSET);
         emit OracleEnabledChanged(enabled);
     }
 
+    function isOracleEnabled() public view returns (bool) {
+        return _miscData.decodeBool(_MISC_ORACLE_ENABLED_OFFSET);
+    }
+
     // Caller must be approved by the Vault's Authorizer
     function setPaused(bool paused) external authenticate {
         _setPaused(paused);
     }
 
+    function _isOwnerOnlyAction(bytes32 actionId) internal view override returns (bool) {
+        // This implementation adds support for the enableOracle action identifier.
+        return super._isOwnerOnlyAction(actionId) || actionId == getActionId(WeightedPool2Tokens.enableOracle.selector);
```

Comment:
The requirement is that governance can _always_ enable the oracle. For simplicity's sake, I thought of not letting the owner do this (which means governance can) - it seems like a weird permission for the owner regardless. If they want it on, they can always turn it on at creation.

(meaning I'd remove this override)